ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/ToExpr.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- | This module implements QSM's @CanDiff@ typeclass using @tree-diff@'s
+-- @ToExpr@.
+module Test.Util.ToExpr () where
+
+import           Data.TreeDiff as T
+import qualified Test.StateMachine as QSM
+import           Test.StateMachine.Diffing (CanDiff (..))
+import qualified Test.StateMachine.Types.References as QSM
+
+instance ToExpr x => CanDiff x where
```

Comment:
Would it make sense to add more comments on this typeclass, and perhaps explain how it is related to `ToExpr`?

Reply:
Well, this is the implementation. The comments live in https://github.com/stevana/quickcheck-state-machine/blob/00db0b4ebcf4faef434b98b0caaeeba0f17002b9/src/Test/StateMachine/Diffing.hs

Reply:
I think the module header should point the reader in the right direction to visit that other file
```
-- | This module implements QSM's @CanDiff@ typeclass using @tree-diff@'s
-- @ToExpr@.
```

Reply:
Good point. I missed that.